pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderTls.java

```diff
@@ -19,15 +19,66 @@
 package org.apache.pulsar.broker.authentication;
 
 import java.io.IOException;
+import java.security.PublicKey;
 import java.security.cert.Certificate;
 import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPublicKey;
+import java.time.*;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
 
 import javax.naming.AuthenticationException;
+import javax.security.auth.x500.X500Principal;
 
+import io.prometheus.client.Counter;
+import lombok.NonNull;
 import org.apache.pulsar.broker.ServiceConfiguration;
 import org.apache.pulsar.broker.authentication.metrics.AuthenticationMetrics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class AuthenticationProviderTls implements AuthenticationProvider {
+    private static final Logger LOG = LoggerFactory.getLogger(AuthenticationProviderTls.class);
+
+    static final Counter clientCertSelfSignedMetrics = Counter.build()
```

Comment:
Could you please move to `broker/stats` dir? all of the metrics components are maintained under the package `org.apache.pulsar.broker.stats`

Reply:
I don't think I can move to that package under the pulsar-broker without creating a circular dependency between pulsar-broker and pulsar-broker-common.

I did move the stats out of the AuthenticationProviderTls class and into the existing AuthenticationMetrics class which was created in [9244](https://github.com/apache/pulsar/pull/9244) so it's in the same place as all the other auth metrics.